File: client/src/shared/enums.ts
export enum FormRenderStateEnum {
  EDIT = 'EDIT',
  VIEW = 'VIEW',
  FIRST_SUBMIT = 'FIRST_SUBMIT',
  SUBMIT_TEMPLATE = 'SUBMIT_TEMPLATE',
  VIS_COND = 'VIS_COND',
  VIS_COND_DISABLED = 'VIS_COND_DISABLED',
}

export enum SexEnum {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
}

export enum GestationalAgeUnitEnum {
  WEEKS = 'WEEKS',
  MONTHS = 'MONTHS',
}

export enum StatsOptionEnum {
  THIS_YEAR = 'THIS_YEAR',
  LAST_TWELVE_MONTHS = 'LAST_TWELVE_MONTHS',
}

export enum TrafficLightEnum {
  GREEN = 'GREEN',
  YELLOW_UP = 'YELLOW_UP',
  YELLOW_DOWN = 'YELLOW_DOWN',
  RED_UP = 'RED_UP',
  RED_DOWN = 'RED_DOWN',
  //original 'UNAVAILABLE' now changed to 'NONE',
  NONE = 'NONE',
}

export enum UserRoleEnum {
  VHT = 'VHT',
  CHO = 'CHO',
  HCW = 'HCW',
  ADMIN = 'ADMIN',
}

export enum MethodEnum {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  DELETE = 'DELETE',
}

export enum EndpointEnum {
  ASSESSMENTS = '/assessments',
  HEALTH_FACILITIES = '/facilities',
  MEDICAL_HISTORY = '/medical_history' /* /patients/{PATIENT_ID}/medicalHistory */,
  MEDICAL_INFO = '/medical_info',
  MEDICAL_RECORDS = '/medical_records',
  PATIENTS = '/patients',
  PATIENT_INFO = '/info' /* /patients/{PATIENT_ID}/info */,
  PATIENT_TIMELINE = '/timeline' /* /patients/{PATIENT_ID}/timeline */,
  PREGNANCIES = '/pregnancies',
  PREGNANCY_RECORDS = '/pregnancies' /* /patients/{PATIENT_ID}/pregnancies */,
  PREGNANCY_STATUS = '/pregnancies/status',
  PREGNANCY_SUMMARY = '/pregnancy_summary',
  REFERRALS = '/referrals',
  REFERRALS_ASSESS = '/referrals/assess',
  FORM_TEMPLATES = '/forms/templates',
  FORM_CLASSIFICATIONS = '/forms/classifications',
  READINGS = '/readings',
  READING_ASSESSMENT = '/patients/reading-assessment', //Create a new reading and assessment
  RELAY_SERVER_PHONE = '/relay/server/phone',
  STATISTICS = '/stats',
  STATS_ALL = '/stats/all',
  STATS_FACILITY = '/stats/facility',
  STATS_USER = '/stats/user',
  STATS_USER_EXPORT = '/stats/export',
  ALL_VHTS = '/user/vhts',
  AUTH = '/user/auth',
  REFRESH = '/user/auth/refresh_token',
  UPLOAD_ADMIN = '/upload/admin',
  USER = '/user/',
  USER_ALL = '/user/all',
  USER_CURRENT = '/user/current',
  USER_REGISTER = '/user/register',
  USER_VHTS = '/user/vhts',
  CHANGE_PASS = '/user/current/change_pass',
  RESET_PASS = '/change_pass',
  FORM = '/forms/responses',
  SECRETKEY = '/smskey' /* /user/{USER_ID}/smsKey */,
  //workflow instance
  WORKFLOW_INSTANCES = '/workflow/instances',
  //workflow templates
  WORKFLOW_TEMPLATES = '/workflow/templates',
  WORKFLOW_CLASSIFICATIONS = '/workflow/classifications',
}

export enum QRelationEnum {
  LARGER_THAN = 'LARGER_THAN',
  SMALLER_THAN = 'SMALLER_THAN',
  EQUAL_TO = 'EQUAL_TO',
  CONTAINS = 'CONTAINS',
}

export enum QuestionTypeEnum {
  INTEGER = 'INTEGER',
  STRING = 'STRING',
  MULTIPLE_CHOICE = 'MULTIPLE_CHOICE',
  MULTIPLE_SELECT = 'MULTIPLE_SELECT',
  DATE = 'DATE',
  DATETIME = 'DATETIME',
  CATEGORY = 'CATEGORY',
}

export enum AnswerTypeEnum {
  NUM = 'number',
  TEXT = 'text',
  MC_ID_ARRAY = 'mcIdArray',
  COMMENT = 'comment',
  CATEGORY = '',
}

export enum SecretKeyMessage {
  NORMAL = 'NORMAL',
  EXPIRED = 'EXPIRED',
  WARN = 'WARN',
  NOTFOUND = 'NOTFOUND',
}

//WORKFLOW_INSTANCES STATUS
export enum WorkflowInstanceStatus {
  DRAFT = 'draft',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
}

File: server/enums.py
from enum import Enum

# To add a table to db, make a new class
# create a migration: flask db migrate
# apply the migration: flask db upgrade

#
# ENUMS CLASSES
#


class EnumWithList(Enum):
    @classmethod
    def listNames(cls):
        return [e.name for e in cls]

    @classmethod
    def listValues(cls):
        return [e.value for e in cls]


class StrEnum(str, Enum):
    def __str__(self) -> str:
        return self.value


class RoleEnum(StrEnum):
    VHT = "VHT"
    HCW = "HCW"
    ADMIN = "ADMIN"
    CHO = "CHO"


class SexEnum(StrEnum):
    MALE = "MALE"
    FEMALE = "FEMALE"
    OTHER = "OTHER"


class TrafficLightEnum(StrEnum):
    NONE = "NONE"
    GREEN = "GREEN"
    YELLOW_UP = "YELLOW_UP"
    YELLOW_DOWN = "YELLOW_DOWN"
    RED_UP = "RED_UP"
    RED_DOWN = "RED_DOWN"


class FacilityTypeEnum(StrEnum):
    HCF_2 = "HCF_2"
    HCF_3 = "HCF_3"
    HCF_4 = "HCF_4"
    HOSPITAL = "HOSPITAL"


class QuestionTypeEnum(EnumWithList):
    INTEGER = "INTEGER"
    DECIMAL = "DECIMAL"
    STRING = "STRING"
    MULTIPLE_CHOICE = "MULTIPLE_CHOICE"
    MULTIPLE_SELECT = "MULTIPLE_SELECT"
    DATE = "DATE"
    TIME = "TIME"
    DATETIME = "DATETIME"
    CATEGORY = "CATEGORY"


class QRelationalEnum(StrEnum):
    LARGER_THAN = "LARGER_THAN"
    SMALLER_THAN = "SMALLER_THAN"
    EQUAL_TO = "EQUAL_TO"
    CONTAINS = "CONTAINS"


class ContentTypeEnum(EnumWithList):
    JSON = "application/json"
    CSV = "text/csv"


class HTTPMethodEnum(StrEnum):
    GET = "GET"
    POST = "POST"
    PUT = "PUT"
    DELETE = "DELETE"


class WorkflowStatusEnum(StrEnum):
    ACTIVE = "Active"
    COMPLETED = "Completed"
    CANCELLED = "Cancelled"

class WorkflowStepStatusEnum(StrEnum):
    PENDING = "Pending"
    ACTIVE = "Active"
    COMPLETED = "Completed"
    CANCELLED = "Cancelled"
